.github/workflows/test.yml

```diff
@@ -1,13 +1,67 @@
-name: Test
+name: test
 
 on:
+  pull_request_target:
   push:
     branches:
       - main
-  pull_request:
 
 jobs:
+  ruff:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: chartboost/ruff-action@v1
+  deploy:
+    needs: [ruff]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
what happens on merge to main?

Reply:
Ah my if must have gotten rebased out, nice catch